File: Producto/frontend/src/remote/DTO/OrdenCompraItemDTO.js
import { HCItemOCItemLinkDTO } from './HCItemOCItemLinkDTO';

/**
 * DTO para representar un ítem de Orden de Compra.
 * Espejo de OrdenCompraItemDTO.java del backend.
 */
export class OrdenCompraItemDTO {
    constructor(data = {}) {
        this.idOCItem = data.idOCItem ?? null;
        this.ocId = data.ocId ?? null;
        this.tipoInsumo = data.tipoInsumo ?? '';
        this.insumoId = data.insumoId ?? null;
        this.nombreInsumo = data.nombreInsumo ?? '';
        this.cantidadRequerida = data.cantidadRequerida ?? 0;
        this.cantidadStock = data.cantidadStock ?? 0;
        this.cantidadComprada = data.cantidadComprada ?? 0;
        this.precioUnitario = data.precioUnitario ?? 0;
        this.subtotal = data.subtotal ?? 0;
        this.hcLinks = Array.isArray(data.hcLinks)
            ? data.hcLinks.map(l => new HCItemOCItemLinkDTO(l))
            : [];
    }
}

File: Producto/backend/src/main/java/backend/com/produccion/application/dto/OrdenCompraItemDTO.java
package backend.com.produccion.application.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class OrdenCompraItemDTO {

    private Long idOCItem;
    private Long ocId;
    private String tipoInsumo;
    private Integer articuloId;
    private String nombreInsumo;
    private BigDecimal cantidadRequerida;
    private BigDecimal cantidadStock;
    private BigDecimal cantidadComprada;
    private BigDecimal precioUnitario;
    private BigDecimal subtotal;
    private List<HCItemOCItemLinkDTO> hcLinks;
}
